src/pacstall.py

```diff
@@ -59,61 +62,88 @@ def _format_action_invocation(self, action) -> str:
-)
-commands.add_argument(
-    "-L", "--list", action="store_true", help="list installed packages"
-)
-commands.add_argument("-Up", "--upgrade", action="store_true", help="upgrade packages")
-commands.add_argument(
-    "-Qi", "--query-info", metavar="package", nargs=1, help="show package info"
-)
-
-modifiers.add_argument(
-    "-P",
-    "--disable-prompts",
-    dest="disable_prompts",
-    action="store_true",
-    help="disable prompts for unattended operations",
-)
-modifiers.add_argument(
-    "-K",
-    "--keep",
-    dest="keep",
-    action="store_true",
-    help="retain build directory after installation",
-)
-
-if len(sys.argv) == 1:
-    parser.print_help()
-    sys.exit(1)
-
-parser.parse_args()
+def parse_arguments() -> Namespace:
+    """Parses command line arguments passed to Pacstall.
+
+    Prints help and exits if no argument is passed.
+
+    Returns
+    -------
+    Namespace: Containing all the parsed arguments
+    """
+    parser = ArgumentParser(prog="pacstall", formatter_class=CustomHelpFormatter)
+    commands = parser.add_argument_group("commands").add_mutually_exclusive_group()
+    modifiers = parser.add_argument_group("modifiers")
+
+    commands.add_argument(
+        "-I", "--install", metavar="package", nargs="+", help="install packages"
+    )
+    commands.add_argument(
+        "-S", "--search", metavar="package", nargs="?", help="search for packages"
+    )
+    commands.add_argument(
+        "-R", "--remove", metavar="package", nargs="?", help="remove packages"
+    )
+    commands.add_argument(
+        "-D", "--download", metavar="package", nargs="?", help="download pacscripts"
+    )
+    commands.add_argument(
+        "-A", "--add-repo", metavar="repo", nargs="?", help="add repos to the repo list"
+    )
+    commands.add_argument(
+        "-V",
+        "--version",
+        action="version",
+        version=f"{Foreground.BIBLUE}Pacstall {Foreground.BIWHITE}2.0 {Foreground.BIYELLOW}Kournikova",
+        help="show version",
+    )
+    commands.add_argument(
+        "-L", "--list", action="store_true", help="list installed packages"
+    )
+    commands.add_argument(
+        "-Up", "--upgrade", action="store_true", help="upgrade packages"
+    )
+    commands.add_argument(
+        "-Qi", "--query-info", metavar="package", nargs=1, help="show package info"
+    )
+
+    modifiers.add_argument(
+        "-P",
+        "--disable-prompts",
+        dest="disable_prompts",
+        action="store_true",
+        help="disable prompts for unattended operations",
+    )
+    modifiers.add_argument(
+        "-K",
+        "--keep",
+        dest="keep",
+        action="store_true",
+        help="retain build directory after installation",
+    )
+
+    if len(sys.argv) == 1:
+        parser.print_help()
+        sys.exit(1)
+
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    if getuser() == "root":
+        message.fancy("error", "Pacstall can't be run as root")
```

Comment:
No contractions: `Pacstall cannot be run as root`

Reply:
Okay. But you should have suggested this change when I merged the PR related to this... Also use the suggest button.